reana_server/status.py

```diff
@@ -377,6 +378,21 @@ def get_unschedulable_nodes(self):
         )
         return [node.metadata.name for node in nodes.items]
 
+    def get_nodes_memory(self):
+        """Get list of all node memory capacities."""
+        try:
+            nodes = current_k8s_corev1_api_client.list_node()
+            return [
+                kubernetes_memory_to_bytes(node.status.capacity["memory"])
+                for node in nodes.items
+            ]
+        except ApiException:
+            return [kubernetes_memory_to_bytes("30Gi")]
```

Comment:
Suggestions are welcomed how to improve this part. Since we need some value for total cluster memory capacity, I simply return this dummy value on error. If we don't return any value here workflow priorities could not be set.

Reply:
Not sure what's the best solution.. It's a pity we cannot rely on k8s API... cc/ @tiborsimko

Reply:
Option A: One technique would be to return "0 GiB", which would mean "cannot detect", which would force to put all workflows into requeuing until we can detect.  This is not ideal, but could relieve troubles in case the API error occurs only infrequently.

Option B: We should probably memoise the results of `get_available_memory()` in a module-wide variable, and, if this exception occurs, we can take the last-detected value as a good approximation of what the free memory might look like.  Not ideal, but better than a fixed amount.

I guess B would be preferable over A.

Reply:
Does option B have to be implemented as part of this PR?

Reply:
Remade the logic to fall back to FIFO strategy in case any errors like this occurs and added couple of `#FIXME`ies to adopt the code later to a new Kubernetes release